src/main/java/seedu/taassist/logic/commands/DeletecCommand.java

```diff
@@ -52,10 +52,17 @@ public CommandResult execute(Model model) throws CommandException {
                 nonExistentClasses.add(moduleClass);
             }
         }
+
+        boolean isDeletingFocusedClass = model.isInFocusMode()
+                && existingClasses.stream().anyMatch(c -> c.isSame(model.getFocusedClass()));
+        if (isDeletingFocusedClass) {
```

Comment:
Should this code block be handled in `model` instead? However, I understand it can be quite challenging with the design of `CommandResult`.